src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1618,6 +1651,58 @@ private async Task WriteComplexAndExpandedNavigationPropertyAsync(IEdmProperty e
             }
         }
 
+        private void WriteNestedNavigationProperties(SelectExpandNode selectExpandNode, ResourceContext resourceContext, ODataWriter writer)
+        {
+            Contract.Assert(resourceContext != null);
```

Comment:
Why `Contract.Assert` and not `Debug.Assert` in this private method?

Reply:
I was just consistent with what is in other private methods of the `ODataResourceSerializer`.

Reply:
Updated to `Debug.Assert`